### PR TITLE
Added `user_attrs` to print by optuna studies in `cli.py`

### LINKED ISSUE #3899: Print `study.user_attr` by `optuna studies` command
### Motivation

Using the cli to query studies for their user_attr would be quite useful.  There doesn't seem to be a way to do this, though there does seem to be a way to set a user_attr from the cli

### Description

optuna studies could have a longform option, with format json/yaml/table; or a new option, optuna study --study-name could be added.

### Alternatives (optional)

_No response_

### Additional context (optional)

_No response_

### PR DESCRIPTION
**Pull request by gonzaload**
## Motivation

Resolve #3899

### CODE REVIEW

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. Could you check the values of `user_attrs` in the unit tests, please?

I think we can check them in `test_studies_command` and `test_studies_command_flatten`.
https://github.com/optuna/optuna/blob/1ab1132ea45384a7b2dfbdadb877d8577c47899b/tests/test_cli.py#L273-L392

**Comment by gonzaload** [comment]
It should now be ready

**Comment by toshihikoyanase** [comment]
Thanks! I'll check it.

File: tests/test_cli.py [hunk 1]
```diff
@@ -351,15 +351,17 @@ def test_studies_command_flatten(output_format: Optional[str]) -> None:
         if output_format is None or output_format == "table":
             expected_keys_1 = expected_keys_2 = [
                 "name",
+                "user_attrs",
```
Comment (toshihikoyanase) [inline tests/test_cli.py:354]: Thank you for fixing the test.

I think we can add assertions to check if the values of `user_attrs` are correct. For example, we keep `study_1` empty (no user_attrs) and add 
 a user attribute to `study_2`. Then, we check the values in a similar way to `studies[0]["n_trials"]` and `studies[0]["direction"]`.

What do you think?

File: optuna/cli.py [hunk 2]
```diff
@@ -343,6 +343,7 @@ class _Studies(_BaseCommand):
 
     _study_list_header = [
         ("name", ""),
+        ("user_attrs", ""),
```
Comment (toshihikoyanase) [inline optuna/cli.py:346]: According to https://github.com/optuna/optuna/issues/3899#issuecomment-1232431835, the purpose of this PR is to make `optuna studies` and `optuna trial` consistent in terms of `user_attrs`.
On the other hand, current implementation is inconsistent when the `user_attrs` are empty.

*optuna trial*
`user_attrs` column is hidden when no trials have `user_attrs`.

```python
import optuna

study = optuna.create_study(
    storage="sqlite:///4129.db",
    study_name="4129",
)
```

```console
% study.optimize(objective, n_trials=2)
+--------+-------+---------------------+---------------------+----------------+----------+
| number | value | datetime_start      | datetime_complete   | duration       | state    |
+--------+-------+---------------------+---------------------+----------------+----------+
|      0 |   1.0 | 2022-11-06 14:25:54 | 2022-11-06 14:25:54 | 0:00:00.003276 | COMPLETE |
+--------+-------+---------------------+---------------------+----------------+----------+
```

*optuna studies*
`user_attrs` is always shown.

```console
% optuna create-study --storage sqlite:///4129-empty.db --study-name empty 
%  optuna studies --storage sqlite:///4129-empty.db                                                                                                                   [~/pfn/code/optuna]
+-------+------------+---------------+----------+----------------+
| name  | user_attrs | direction     | n_trials | datetime_start |
+-------+------------+---------------+----------+----------------+
| empty | {}         | ('MINIMIZE',) |        0 |                |
+-------+------------+---------------+----------+----------------+
```

Do we align the behavior when the `user_attrs` values are empty?
@g-votte @nzw0301

Comment (g-votte) [reply optuna/cli.py:346]: As @toshihikoyanase mentions, the goal is to make `optuna studies` and `optuna trials` consistent. It is the case even for the edge case where `user_attrs` is empty.

Comment (gonzaload) [reply optuna/cli.py:346]: True I will fix this

**Comment by toshihikoyanase** [comment]
@knshnb  This PR is also about CLI like #4100. Could you join the review, please?

**Comment by gonzaload** [comment]
So I have manage to emulate the trials behaviour, but I would like to know the final scope of the PR.

So far I have done
- Modify `_Studies(_BaseCommand)` in `cli.py` to print `user_attrs` whenever they are present.
- Modify `test_studies_command` and `test_studies_command_flatten` to test new functionality

After looking at how the `optuna trials` actually work  I realised that the current solution is more of a workaround,
since the functionality to print `user_attrs` is actually implemented in `_convert_to_dict`.
Also I found that currently there is no test in `optuna trials` for checking the print out of `user_attrs` 

So now my question to @g-votte and @toshihikoyanase 
Should this PR also include:
- Delete the workaround and implement a print `user_attrs` functionality based on `_convert_to_dict`.
- Testing of `user_attrs` for `optuna_trials`.
or would the current work be enough and I should create a separate issue for such modifications?

**Comment by toshihikoyanase** [comment]
@gonzaload Thank you for your update. The current workaround is simple, and it seems reasonable to me.
As you mentioned, the current code is a bit inconsistent with `optuna trials` command, but the tasks you suggested seems to be out-of-scope of this PR.

> Delete the workaround and implement a print `user_attrs` functionality based on `_convert_to_dict`: needs quite more time
> Testing of `user_attrs` for optuna_trials.

The first task is a refactoring task and it does not affect users. So, we may be work on it after merging this PR.
About the second task, we can just focus on `optuna studies` test to keep the PR simple.

By the way, I'm a bit busy this week, so let me remove the review assignment.
@c-bata Could you assign a new reviewer, please?

**Comment by c-bata** [comment]
Sure 👍  Let me assign @contramundum53 as a second reviewer.

**Review by contramundum53 — APPROVED** [review]
LGTM!

**Review by knshnb — APPROVED** [review]
Sorry for the delayed response. Let's make the two tasks @toshihikoyanase mentioned follow-up work. LGTM.

**Comment by knshnb** [comment]
@gonzaload Thank you for the contribution to Optuna! Let me merge this PR.

**Pull request merged**